project/ParadoxSupport.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka
+
+import _root_.io.github.lukehutch.fastclasspathscanner.FastClasspathScanner
+import com.lightbend.paradox.markdown._
+import com.lightbend.paradox.markdown.ExternalLinkDirective.LinkException
+import com.lightbend.paradox.sbt.ParadoxPlugin.autoImport._
+import org.pegdown.Printer
+import org.pegdown.ast._
+import sbt.Keys._
+import sbt._
+
+import scala.collection.JavaConverters._
+
+object ParadoxSupport {
+  val paradoxWithCustomDirectives = Seq(
+    paradoxDirectives ++= Def.taskDyn {
+      val classpath = (fullClasspath in Compile).value.files.map(_.toURI.toURL).toArray
+      val classloader = new java.net.URLClassLoader(classpath, this.getClass().getClassLoader())
+      lazy val scanner = new FastClasspathScanner("akka").addClassLoader(classloader).scan()
+      val allClasses = scanner.getNamesOfAllClasses.asScala.toVector
+      Def.task { Seq(
+        { context: Writer.Context ⇒ new UnidocDirective(allClasses, context.location.tree.label, context.properties) }
+      )}
+    }.value
+  )
+
+  class UnidocDirective(allClasses: IndexedSeq[String], page: Page, variables: Map[String, String]) extends InlineDirective("unidoc") {
+    def render(node: DirectiveNode, visitor: Visitor, printer: Printer): Unit = {
+      if (node.label.contains('.')) {
+        val fqcn = node.label
+        val label = fqcn.split('.').last
+        val source = new DirectiveNode.Source.Direct(fqcn)
+        val attributes = new org.pegdown.ast.DirectiveAttributes.AttributeMap()
+        val updatedNode = new DirectiveNode(DirectiveNode.Format.Inline, node.name, label, source, attributes, null,  new TextNode(label))
+        renderByFqcn(updatedNode, visitor, printer)
+      }
+      else {
+        renderByClassName(node.label, node, visitor, printer)
+      }
+    }
+
+    def renderByClassName(label: String, node: DirectiveNode, visitor: Visitor, printer: Printer): Unit = {
```

Comment:
It would be neat if this logic would just find the fqcn based on the short text and then re-use `renderByFqcn`, would that be possible?

Reply:
I iterated further on this a bit in https://github.com/akka/akka-http/pull/1890, could you have a look whether what I did there make sense and if so apply the same here?

Reply:
Sure, copied `UnidocDirective` from akka/akka-http#1890